nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -869,7 +870,7 @@ in
             if cfgScrub.pools != [] then
               (lib.concatStringsSep " " cfgScrub.pools)
             else
-              "$(${cfgZfs.package}/bin/zpool list -H -o name)"
+              "\"$(${cfgZfs.package}/bin/zpool list -H -o name)\""
```

Comment:
This is not correct. This is passing all pool names as a single argument. That one argument is a string with newlines between each pool name, but it's one argument and `zpool scrub` will not accept it.